src/app/campaign/flight/flight.container.ts

```diff
@@ -85,4 +98,24 @@ export class FlightContainerComponent implements OnInit, OnDestroy {
     this.campaignStoreService.loadAvailability(flight);
     this.currentInventoryUri$.next(flight.set_inventory_uri);
   }
+
+  flightDuplicate(flight: Flight) {
+    const localFlight: Flight = { ...flight, name: `${flight.name} (Copy)` };
+    this.campaignStoreService.campaignFirst$.subscribe(state => {
+      const flightId = Date.now();
```

Comment:
should we encapsulate this temp flightId generation in the campaignStoreService?